cluster/src/main/java/org/apache/iotdb/cluster/utils/ClusterUtils.java

```diff
@@ -353,4 +361,65 @@ public static PartitionGroup partitionByPathTimeWithSync(
     }
     return partitionGroup;
   }
+
+  public static void setVersionForSpecialPlan(PhysicalPlan plan, long currLogIndex)
+      throws MetadataException {
+    switch (plan.getOperatorType()) {
+      case SET_STORAGE_GROUP:
+        plan.setMajorVersion(currLogIndex);
+        plan.setMinorVersion(0);
+        break;
+      case DELETE_STORAGE_GROUP:
+        List<PartialPath> deletePathList = new ArrayList<>();
+        for (PartialPath storageGroupPath : plan.getPaths()) {
+          List<PartialPath> allRelatedStorageGroupPath =
+              IoTDB.metaManager.getStorageGroupPaths(storageGroupPath);
+          if (allRelatedStorageGroupPath.isEmpty()) {
+            throw new PathNotExistException(storageGroupPath.getFullPath(), true);
+          }
+          for (PartialPath path : allRelatedStorageGroupPath) {
+            StorageGroupMNode storageGroupMNode = IoTDB.metaManager.getStorageGroupNodeByPath(path);
+            path.setMajorVersion(storageGroupMNode.getMajorVersion());
+            path.setMinorVersion(storageGroupMNode.getMinorVersion());
+          }
+        }
+        // replace the to be deleted paths
+        plan.setPaths(deletePathList);
+        break;
```

Comment:
Maybe it is not necessary to record the major version of every deleted SGs as we can delete the SG as long as its version is smaller than that of the plan.